nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ReplaceText.java

```diff
@@ -287,14 +287,24 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
 
         if (evaluateMode.equalsIgnoreCase(ENTIRE_TEXT)) {
             if (flowFile.getSize() > maxBufferSize && replacementStrategyExecutor.isAllDataBufferedForEntireText()) {
-                session.transfer(flowFile, REL_FAILURE);
+                session.transfer(session.penalize(flowFile), REL_FAILURE);
```

Comment:
@mosermw are you penalizing it here in case someone is looping failure?  If they loop failure relationship then the problem will happen over and over anyway and perhaps better we just doc that they should not do so.  Were you thinking a downstream consumer of this would prefer it to be penalized?  Just trying to understand the benefit of penalizing here.